pycqed/measurement/multi_qubit_module.py

```diff
@@ -447,6 +449,256 @@ def measure_multiplexed_readout(qubits, liveplot=False,
             use_preselection=preselection
         ))
 
+def measure_ssro(qubits, states=('g', 'e'), n_shots=10000, label=None,
+                 preselection=True, all_states_combinations=False, upload=True,
```

Comment:
Shouldn't measure_ssro and measure_optimized weights use the Device object?

Reply:
It could but I'm not sure I would understand for what exactly it would be used as those measurement function do not use anything related to two-qubit gates. To my understanding, this was the main use-case of the device object so far (?). Or did we agree that all functions should use the device object no matter whether it uses it or not? I could see in the future that we for instance add a config to the device object that maybe specifies which qubit can be calibrated simultaneously etc., which might be relevant when we start automating calibration etc. but for now I guess the user in the notebook is still taking care of that. In any case, whether we do it now or later, it will be simple to just add the device object as a parameter to this function

Reply:
As i understood it, we would like to use the device object for all multi-qubit algorithms. I also added it to the multi-qubit time domain as per Ants' suggestion. Otherwise we create a mess if only some functions use it :p 
It has wrappers around get_operation_dict and get_prep_params which should become the new standard to use for all multi-qubit measurements.

Reply:
And the device object stores all qubits objects. I agreed with Michael that the measure functions should then accept qubit names as well as qubits objects:
`qubits = [dev.get_qb(qb) if isinstance(qb, str) else qb for qb in qubits]`

I think there is no need to modify old functions if they do not use 2qb gates. But for new multi-qubit functions, it might make sense to already stick to this concept, even if they do not need 2qb gates. What do you think?

Reply:
Ok for me, I will then add this now.